### PR TITLE
web: Add a button to opt into the new UX

### PR DESCRIPTION
**Pull request by nicks**
Hello @hyu,

Please review the following commits I made in branch nicks/ch10712:

a7b362d4d6dfcded2a468f57e7791efa7e874bb8 (2021-01-11 20:11:49 -0500)
web: Add a button to opt into the new UX

55105dab830a7913de1fe505daf1ba2c775f1f8b (2021-01-11 20:07:26 -0500)
web: add an update dialog. Also fixes https://github.com/tilt-dev/tilt/issues/3844 while I'm here

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

**Comment by nicks** [comment]
Screenshot from 2021-01-11 20-08-53
Screenshot from 2021-01-11 20-08-38

I felt like the interaction of navigating what you hit the switch was too abrupt, so i added a refresh link that appears when you're in the old ui but have the new ui on by default

File: web/src/UpdateDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+      checked={isNewDefault}
+      onChange={interfaceVersion.toggleDefault}
+    />
+  )
+  let label = isWrongInterface ? (
+    <div>
+      See Tilt in the new UI (<Link to={pathBuilder.path("/")}>Refresh</Link>)
+    </div>
+  ) : (
+    <div>See Tilt in the new UI</div>
```
Comment (hyu) [inline web/src/UpdateDialog.tsx:104]: Can we try "Preview Tilt's new interface"? (Assuming that it's more common terminology than "UI")

File: web/src/UpdateDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+      </span>
+    </div>
+  ) : null
+
+  let isNewDefault = interfaceVersion.isNewDefault()
+  let isWrongInterface = isNewDefault != props.isNewInterface
+  let title =
+    !props.isNewInterface || props.showUpdate
+      ? "Updates available"
+      : "Back to the old UI"
```
Comment (hyu) [inline web/src/UpdateDialog.tsx:92]: Maybe "Tilt Interface Setting" instead for this title? (To complement my suggestion for the toggle label)

**Review by hyu — APPROVED** [review]

File: web/src/UpdateDialog.tsx (hunk 1 above)
Comment (nicks) [reply web/src/UpdateDialog.tsx]: done

File: web/src/UpdateDialog.tsx (hunk 2 above)
Comment (nicks) [reply web/src/UpdateDialog.tsx]: ya!

**Pull request merged**